### ISSUE TITLE
Cannot create files with special characters in filename on Windows volume mount (grpc fuse)

### ISSUE THREAD

**Issue by abarani** [docker/for-win#5520]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Expected behavior
```
$ echo "test" > "1:2:3.txt"
```
File is created correctly.

### Actual behavior
```
$ echo "test" > "1:2:3.txt"
bash: 1:2:3.txt: Input/output error
```
Input/output error during file creation.

### Information

  - Windows Version: Windows 10 Pro, build 18363.592
  - Docker Desktop Version: 2.2.0.0
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: No

### Steps to reproduce the behavior
Create a file using some NTFS reserved character in a Linux container.
Tested with: `* " | : < > ?`

**Comment by stephen-turner** [comment]
I can confirm that this used to work in the previous Samba file sharing solution. I'm not sure how easy it will be for us to implement it in the new Fuse solution though. We'll investigate.

**Comment by stephen-turner** [comment]
We have a fix for this under development.

**Comment by stephen-turner** [comment]
We have a new private build which (we believe) fixes this bug. https://download-stage.docker.com/win/stable/42579/Docker%20Desktop%20Installer.exe. Please try it and let us know how you get on.

**Comment by abarani** [comment]
It seems to be working as expected now.
Thank you for the quick fix!

**Comment by stephen-turner** [comment]
We have just released 2.2.0.3, which fixes this issue, so closing this ticket. In case any Edge users are watching this ticket, it will be on the Edge channel tomorrow too.
